src/adapter/src/lib.rs

```diff
@@ -74,6 +74,10 @@
 #![warn(clippy::from_over_into)]
 // END LINT CONFIG
 #![cfg_attr(nightly_doc_features, feature(doc_cfg))]
+// Without this, cargo clippy complains with:
+//     overflow evaluating the requirement `&str: std::marker::Send`
+// in implement_peek_plan's return of a Result<crate::ExecuteResponse, AdapterError>.
+#![recursion_limit = "256"]
```

Comment:
This is worrisome to me because I don't understand it.

Reply:
me neither

Reply:
This seems to be causing cargo check to take about 25s just for the adapter crate:

```
~/materialize@work (rm-stash)$ cargo check
    Checking mz-adapter v0.0.0 (/home/mjibson/materialize/src/adapter)
    Checking mz-pgwire v0.0.0 (/home/mjibson/materialize/src/pgwire)
    Checking mz-testdrive v0.0.0 (/home/mjibson/materialize/src/testdrive)
    Checking mz-stash-debug v0.28.0-dev (/home/mjibson/materialize/src/stash-debug)
    Checking mz-environmentd v0.40.0-dev (/home/mjibson/materialize/src/environmentd)
    Checking mz-sqllogictest v0.0.1 (/home/mjibson/materialize/src/sqllogictest)
    Finished dev [unoptimized + debuginfo] target(s) in 33.10s
```

(they all went quite fast after adapter)